lib/spack/docs/build_systems/autotoolspackage.rst

```diff
@@ -112,20 +112,47 @@ phase runs:
 
 .. code-block:: console
 
-   $ libtoolize
-   $ aclocal
-   $ autoreconf --install --verbose --force
+   $ autoreconf --install --verbose --force -I <aclocal-prefix>/share/aclocal
+
+In case you need to add more arguments, override ``autoreconf_extra_args``
+in your ``package.py`` on class scope like this:
+
+.. code-block:: python
+
+   autoreconf_extra_args = ["-Im4"]
 
 All you need to do is add a few Autotools dependencies to the package.
 Most stable releases will come with a ``configure`` script, but if you
-check out a commit from the ``develop`` branch, you would want to add:
+check out a commit from the ``master`` branch, you would want to add:
 
 .. code-block:: python
 
-   depends_on('autoconf', type='build', when='@develop')
-   depends_on('automake', type='build', when='@develop')
-   depends_on('libtool',  type='build', when='@develop')
-   depends_on('m4',       type='build', when='@develop')
+   depends_on('autoconf', type='build', when='@master')
+   depends_on('automake', type='build', when='@master')
+   depends_on('libtool',  type='build', when='@master')
+
+When the package has ``autoconf`` as dependenency, and ``m4`` is only used
+as the script interpreter of ``autoconf`` and not for other purposes, it
+does not have to be listed explicitly and can be removedd as it is not needed.
+Spack versions 0.16.x and lower may wrongly terminate their autoreconf phase
+if it is not listed as an explicit dependency.
```

Comment:
```suggestion
```
I think we can skip this and target the docs to (new) users of the current version of Spack. In fact `m4` is not mentioned anywhere in this page (well, except in the `autoreconf_extra_args` example). If people add `m4` anyways we can catch it during review of their PR.

Reply:
Done, rebased to develop and now it is only a doc PR. Removed the other labels. Ready for review.